src/Fhir.Anonymizer.AzureDataFactoryPipeline/src/FhirBlobConsumer.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Fhir.Anonymizer.Core;
+
+namespace Fhir.Anonymizer.AzureDataFactoryPipeline.src
+{
+    public class FhirBlobConsumer : IFhirDataConsumer
+    {
+        private static byte[] _newLine = Encoding.UTF8.GetBytes("\r\n");
+
+        private BlockBlobClient _blobClient;
+        private List<Task> _runningTasks;
+        private List<string> _blockIds;
+        private MemoryStream _currentStream;
+
+        public FhirBlobConsumer(BlockBlobClient blobClient)
+        {
+            _blobClient = blobClient;
+
+            _runningTasks = new List<Task>();
+            _blockIds = new List<string>();
+            _currentStream = new MemoryStream();
+        }
+
+        public int UploadBlockThreshold
+        {
+            get;
+            set;
+        } = FhirAzureConstants.DefaultUploadBlockThreshold;
+
+        public int ConcurrentCount
+        {
+            get;
+            set;
+        } = FhirAzureConstants.DefaultConcurrentCount;
+
+        public IProgress<long> ProgressHandler
+        {
+            set;
+            private get;
+        } = null;
+
+        public int BlockUploadTimeoutInSeconds
+        {
+            get;
+            set;
+        } = FhirAzureConstants.DefaultBlockUploadTimeoutInSeconds;
+
+        public int BlockUploadTimeoutRetryCount
+        {
+            get;
+            set;
+        } = FhirAzureConstants.DefaultBlockUploadTimeoutRetryCount;
+
+        public async Task<int> ConsumeAsync(IEnumerable<string> data)
+        {
+            int result = 0;
+            foreach (string item in data)
+            {
+                byte[] byteData = Encoding.UTF8.GetBytes(item);
+                await _currentStream.WriteAsync(byteData, 0, byteData.Length).ConfigureAwait(false);
+                await _currentStream.WriteAsync(_newLine, 0, _newLine.Length).ConfigureAwait(false);
+                if (_currentStream.Length >= UploadBlockThreshold)
+                {
+                    await AddCurrentStreamToUploadTaskListAsync().ConfigureAwait(false);
+                }
+
+                result++;
+            }
+
+            return result;
+        }
+
+        public async Task CompleteAsync()
+        {
+            if (_currentStream.Length > 0)
+            {
+                await AddCurrentStreamToUploadTaskListAsync().ConfigureAwait(false);
+            }
+
+            await Task.WhenAll(_runningTasks).ConfigureAwait(false);
+            foreach (var task in _runningTasks.FindAll(t => t.IsCompleted))
```

Comment:
Shouldn't all be completed here (FindAll not needed)?

Reply:
Thanks, fixed.